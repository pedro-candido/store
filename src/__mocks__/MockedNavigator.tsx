import React from 'react';

import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();

type Params = {
  component: React.ComponentType<any>;
  [key: string]: any;
};

const MockedNavigator = ({ component, params = {} }: Params) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockedNavigator;
