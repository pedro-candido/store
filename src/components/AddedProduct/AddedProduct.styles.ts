import styled from 'styled-components/native';

import Button from '@components/Button';
import Label from '@components/Label';

export const Container = styled.View`
  flex-direction: row;
  max-width: 100%;
  justify-content: space-around;
  align-items: center;
  margin-top: 2%;
`;

export const Picture = styled.Image`
  width: 50px;
  height: 50px;
`;

export const StyledButton = styled(Button)`
  max-width: 36px;
  margin-top: 0;
`;

export const StyledLabel = styled(Label)`
  width: 50px;
`;
