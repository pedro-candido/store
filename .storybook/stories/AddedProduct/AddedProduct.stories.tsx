import { View } from 'react-native';
import type { Meta, StoryObj } from '@storybook/react';
import AddedProduct from '@/components/AddedProduct';
import { products } from '@/__mocks__/productsMock';

const AddedProductMeta: Meta<typeof AddedProduct> = {
  title: 'AddedProduct',
  component: AddedProduct,
  args: {
    children: 'Hello world',
  },
  decorators: [
    (Story) => (
      <View style={{ justifyContent: 'center', flex: 1 }}>
        <Story />
      </View>
    ),
  ],
};

export default AddedProductMeta;

export const Default: StoryObj<typeof AddedProduct> = {
  args: {
    ...products[0],
  },
};
