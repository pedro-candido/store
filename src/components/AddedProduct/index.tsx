import { Container, Picture, StyledButton } from '@components/AddedProduct/AddedProduct.styles';
import Label from '@components/Label';
import { TProductsChosen } from '@store/products.store';
import { convertNumberToDollar } from '@utils/formatter';

function AddedProduct({ image, price, title }: TProductsChosen) {
  return (
    <Container>
      <Picture source={{ uri: image }} />
      <Label
        numberOfLines={1}
        style={{
          width: 100,
        }}
      >
        {title}
      </Label>
      <Label>{convertNumberToDollar(price)}</Label>
      <StyledButton type={'secondary'} iconName={'close'} />
    </Container>
  );
}

export default AddedProduct;
