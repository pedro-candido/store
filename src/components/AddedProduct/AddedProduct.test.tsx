import React from 'react';

import { fireEvent, render } from '@testing-library/react-native';

import { products } from '@/__mocks__/productsMock';

import AddedProduct from './index';
const { title, image, price, id } = products[0];

const mockedFn = jest.fn();

jest.mock('@store/globalStore', () => ({
  useStore: () => ({
    removeProduct: mockedFn,
  }),
}));

describe('AddedProduct', () => {
  it('should match snapshot', () => {
    const testRenderer = render(
      <AddedProduct productTestId={''} productId={id} image={image} price={price} title={title} />
    );

    expect(testRenderer.toJSON).toMatchSnapshot();
  });
  it('should press on button', () => {
    const buttonTest = 'button-testId';
    const { getByTestId } = render(
      <AddedProduct
        productTestId={buttonTest}
        productId={id}
        image={image}
        price={price}
        title={title}
      />
    );

    const button = getByTestId(buttonTest);

    fireEvent.press(button);

    expect(mockedFn).toHaveBeenCalled();
  });
});
