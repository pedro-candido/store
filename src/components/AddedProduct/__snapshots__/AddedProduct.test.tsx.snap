// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddedProduct should match snapshot 1`] = `[Function]`;
